crate/server/src/core/certificate/parsing.rs

```diff
@@ -23,22 +22,18 @@ use crate::{error::KmsError, result::KResult};
 /// The key identifier in hex encoding
 pub(crate) fn get_crl_authority_key_identifier(
     x509_crl: &CertificateRevocationList<'_>,
-) -> KResult<String> {
-    let aki_option = x509_crl
+) -> Option<String> {
+    match x509_crl
         .extensions()
         .iter()
         .find(|&ext| ext.oid == OID_X509_EXT_AUTHORITY_KEY_IDENTIFIER)
         .and_then(|ext| match ext.parsed_extension() {
             ParsedExtension::AuthorityKeyIdentifier(aki) => Some(aki),
             _ => None,
-        });
-    let aki = aki_option.ok_or(KmsError::Certificate(
-        "Extension Authority Key Identifier not found".to_string(),
-    ))?;
-    let ki = aki.key_identifier.clone().ok_or(KmsError::Certificate(
-        "Authority Key Identifier does not have Key Identifier".to_string(),
-    ))?;
-    Ok(hex::encode(ki.0))
+        }) {
+        Some(aki) => aki.key_identifier.as_ref().map(|ki| hex::encode(ki.0)),
+        None => None,
+    }
```

Comment:
you can `map` if you want

Reply:
thanks done in 423bb86856c2b1d6aadf5e6802414ef2318fecf4